src/modules/logicModule/paymentClient/ERC20PaymentClient.sol

```diff
@@ -208,4 +184,53 @@ abstract contract ERC20PaymentClient is
             revert Module__ERC20PaymentClient__InvalidRecipient();
         }
     }
+
+    //--------------------------------------------------------------------------
+    // {ERC20PaymentClient} Function Implementations
+
+    /// @dev Ensures `amount` of payment tokens exist in address(this).
+    function _ensureTokenBalance(uint amount) internal virtual {
+        uint balance = __Module_orchestrator.token().balanceOf(address(this));
+
+        if (balance < amount) {
+            // Trigger callback from orchestrator to transfer tokens
+            // to address(this).
+            bool ok;
+            (ok, /*returnData*/ ) = __Module_orchestrator.executeTxFromModule(
+                address(__Module_orchestrator.fundingManager()),
+                abi.encodeWithSignature(
+                    "transferOrchestratorToken(address,uint256)",
+                    address(this),
+                    amount - balance
+                )
+            );
+
+            if (!ok) {
+                revert Module__ERC20PaymentClient__TokenTransferFailed();
+            }
+        }
+    }
+
+    /// @dev Ensures `amount` of token allowance for payment processor(s).
+    function _ensureTokenAllowance(IPaymentProcessor spender, uint amount)
+        internal
+        virtual
+    {
+        IERC20 token = __Module_orchestrator.token();
+        uint allowance = token.allowance(address(this), address(spender));
```

Comment:
Not storing these values in separate variables can save gas. 

But it might make the code less neat than it is now.

Your call.